examples/prometheus-kubernetes/main.go

```diff
@@ -0,0 +1,1286 @@
+						},
+						TargetLabel: "__metrics_path__",
+					},
+					{
+						Action: prometheus.RelabelReplace,
+						Regex: prometheus.MustNewRegexp("([^:]+)(?::\\d+)?;(\\d+)"),
+						Replacement: "$1:$2",
+						SourceLabels: model.LabelNames{
+							"__address__",
+							"__meta_kubernetes_service_annotation_prometheus_io_port",
+						},
+						TargetLabel: "__address__",
+					},
+					{
+						Action: prometheus.RelabelLabelMap,
+						Regex: prometheus.MustNewRegexp("__meta_kubernetes_pod_label_(.+)"),
+					},
+					{
+						Action: prometheus.RelabelReplace,
+						SourceLabels: model.LabelNames{
+							"__meta_kubernetes_namespace",
+						},
+						TargetLabel: "kubernetes_namespace",
+					},
+					{
+						Action: prometheus.RelabelReplace,
+						SourceLabels: model.LabelNames{
+							"__meta_kubernetes_pod_name",
+						},
+						TargetLabel: "kubernetes_pod_name",
+					},
+				},
+			},
+		},
+		AlertingConfig:prometheus.AlertingConfig{
+			AlertmanagerConfigs: []*prometheus.AlertmanagerConfig{
+				{
+					ServiceDiscoveryConfig: sdconfig.ServiceDiscoveryConfig{
+						KubernetesSDConfigs: []*kubernetes.SDConfig{
+							{
+								Role: kubernetes.RolePod,
+							},
+						},
+					},
+					HTTPClientConfig: config.HTTPClientConfig{
+						TLSConfig: config.TLSConfig{
+							CAFile: "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt",
+							InsecureSkipVerify: true,
+						},
+						BearerTokenFile: "/var/run/secrets/kubernetes.io/serviceaccount/token",
+					},
+					RelabelConfigs: []*prometheus.RelabelConfig{
+						{
+							Action: prometheus.RelabelKeep,
+							SourceLabels: model.LabelNames{
+								"__meta_kubernetes_namespace",
+							},
+							Regex: prometheus.MustNewRegexp("default"),
+						},
+						{
+							Action: prometheus.RelabelKeep,
+							SourceLabels: model.LabelNames{
+								"__meta_kubernetes_pod_label_app",
+							},
+							Regex: prometheus.MustNewRegexp("prometheus"),
+						},
+						{
+							Action: prometheus.RelabelKeep,
+							SourceLabels: model.LabelNames{
+								"__meta_kubernetes_pod_label_component",
+							},
+							Regex: prometheus.MustNewRegexp("alertmanager"),
+						},
+						{
+							Action: prometheus.RelabelDrop,
+							SourceLabels: model.LabelNames{
+								"__meta_kubernetes_pod_container_port_number",
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+
+	// TODO: Is there a cleaner way of doing this?
+	serverConfigBytes := new(bytes.Buffer)
+	if _, err := serverConfigBytes.ReadFrom(encoding.YAML(serverConfig)); err != nil {
+		panic(err)
+	}
+
+	serverConfigMap := corev1.ConfigMap{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "release-name-prometheus-server",
+			Labels: map[string]string{
+				"app": "prometheus",
+				"component": "server",
+			},
+		},
+		Data: map[string]string{
+			"prometheus.yml": serverConfigBytes.String(),
+		},
+	}
+
+	generator.Add("server-configmap.yaml", encoding.YAML(serverConfigMap))
```

Comment:
Wonder - we should do it at the end or one by one?

Reply:
Hmmm - trying them both out I like doing it all at the end better. That way we are grouping concerns together rather than having them spread out. Easier to reason about.